drivers/mtd/spi_flash.c

```diff
@@ -706,6 +706,114 @@ static ssize_t spi_write(FAR struct mtd_dev_s *dev, off_t offset,
 }
 #endif
 
+int spi_test_rwe(struct mtd_dev_s *dev, uint16_t *nbadblock)
+{
+  int ret = OK;
+  uint8_t *buf;
+  uint16_t value = 0xAA;
+  uint16_t block = 10;
+  uint16_t i = 0;
+  uint16_t badblock_num = 0;
+
+  FAR struct spi_flash_dev_s *priv = (FAR struct spi_flash_dev_s *)dev;
+  off_t offset = block << priv->block_shift;
+  uint16_t pagesperblk = priv->block_size / priv->page_size;
+
+  buf = (FAR uint8_t *)kmm_malloc(priv->page_size);
+  if(!buf) {
+    ferr("malloc buffer error...");
+    kmm_free(buf);
+    ret = 4;
+    goto errout;
+  }
+
+  for(i=0; i<priv->nblocks; i++) {
+    ret = spi_erase(dev, i, 1);
+    if(ret != 1) {
+      if(ret == -EIO) {
+        ferr("Find bad block %d , ret: %d\n", i, ret);
+        badblock_num++;
+      } else {
+        ferr("Erase error, ret: %d\n", ret);
+        ret = 3;
+         goto errout;
+      }
+    }
```

Comment:
These two level of if statements could be merged into one

if (ret == -EIO) {
} else if (ret !=1)
}